lib/logstash/filters/kv.rb

```diff
@@ -41,29 +41,65 @@ class LogStash::Filters::KV < LogStash::Filters::Base
   # These characters form a regex character class and thus you must escape special regex
   # characters like `[` or `]` using `\`.
   #
-  # For example, to strip `<`, `>`, `[`, `]` and `,` characters from values:
+  # Only leading and trailing characters are trimed from the value.
+  #
+  # For example, to trim `<`, `>`, `[`, `]` and `,` characters from values:
   # [source,ruby]
   #     filter {
   #       kv {
-  #         trim => "<>\[\],"
+  #         trim_value => "<>\[\],"
   #       }
   #     }
-  config :trim, :validate => :string
+  config :trim_value, :validate => :string
 
   # A string of characters to trim from the key. This is useful if your
   # keys are wrapped in brackets or start with space.
   #
   # These characters form a regex character class and thus you must escape special regex
   # characters like `[` or `]` using `\`.
   #
-  # For example, to strip `<` `>` `[` `]` and `,` characters from keys:
+  # Only leading and trailing characters are trimed from the key.
+  #
+  # For example, to trim `<` `>` `[` `]` and `,` characters from keys:
+  # [source,ruby]
+  #     filter {
+  #       kv {
+  #         trim_key => "<>\[\],"
+  #       }
+  #     }
+  config :trim_key, :validate => :string
+
+  # A string of characters to remove from the value.
+  #
+  # These characters form a regex character class and thus you must escape special regex
+  # characters like `[` or `]` using `\`.
+  #
+  # Contrary to trim option, all characters are removed from the value, whatever their position.
+  #
+  # For example, to remove `<`, `>`, `[`, `]` and `,` characters from values:
   # [source,ruby]
   #     filter {
   #       kv {
-  #         trimkey => "<>\[\],"
+  #         remove_char_value => "<>\[\],"
   #       }
   #     }
-  config :trimkey, :validate => :string
```

Comment:
Hindsight: This should have been marked deprecated, adding: `:deprecated => "This setting is renamed to 'remove_char_value'"`